### ISSUE TITLE
Media-RSS support

### ISSUE THREAD

**Issue by azmeuk** [Xyrio/RSSOwlnix#74]
Some video platform like Youtube or [Peertube](https://joinpeertube.org) use the [Media-RSS](http://www.rssboard.org/media-rss) extension to provide metadata about the videos. They generally do not use the whole tags available, but provide tags for the video description, the thumbnail, and the number of views.

It would be amazing if RssOwl could at least display the thumbnail and the description for videos.

An extract of a [Youtube feed](https://www.youtube.com/feeds/videos.xml?channel_id=UC7_gcs09iThXybpVgjHZ_7g):

```xml
  <media:group>
   <media:title>Can You Observe a Typical Universe?</media:title>
   <media:content url="https://www.youtube.com/v/q-6oU3jXAho?version=3" type="application/x-shockwave-flash" width="640" height="390"/>
   <media:thumbnail url="https://i2.ytimg.com/vi/q-6oU3jXAho/hqdefault.jpg" width="480" height="360"/>
   <media:description>Sign Up on Patreon to get access to the Space Time Discord! 
https://www.patreon.com/pbsspacetime

Check out the Space Time Merch Store
https://pbsspacetime.com/

The moment you started observing reality, you hopelessly polluted any conclusions you might make about it. The anthropic principle guarantees that you are NOT seeing the universe in most typical state. But used correctly, this highly controversial idea can be extremely powerful. So, how do you correctly use the anthropic principle?

Hosted by Matt O'Dowd
Written by Matt O'Dowd
Graphics by Leonardo Scholzer &amp; Adriano Leal
Directed by: Andrew Kornhaber
Executive Producers: Eric Brown &amp; Andrew Kornhaber 
...</media:description>
   <media:community>
    <media:starRating count="7709" average="4.92" min="1" max="5"/>
    <media:statistics views="176224"/>
   </media:community>http://www.rssboard.org/media-rss
  </media:group>
```

How it is displayed:
![rssowl-youtube](https://user-images.githubusercontent.com/60163/69174287-c8cd6c00-0b01-11ea-9351-20e8a9b7451f.png)

An extract of a [Peertube feed](https://framatube.org/feeds/videos.xml?accountId=3):
```xml


<media:group>
                <media:content type="application/x-bittorrent" url="https://framatube.org/static/torrents/ed5c048d-01f3-4ceb-97db-6e278de512b0-1080.torrent" isDefault="true">
                </media:content>
                <media:content type="application/x-bittorrent" url="https://framatube.org/static/torrents/ed5c048d-01f3-4ceb-97db-6e278de512b0-720.torrent">
                </media:content>
                <media:content type="application/x-bittorrent" url="https://framatube.org/static/torrents/ed5c048d-01f3-4ceb-97db-6e278de512b0-480.torrent">
                </media:content>
                <media:content type="application/x-bittorrent" url="https://framatube.org/static/torrents/ed5c048d-01f3-4ceb-97db-6e278de512b0-360.torrent">
                </media:content>
                <media:content type="application/x-bittorrent" url="https://framatube.org/static/torrents/ed5c048d-01f3-4ceb-97db-6e278de512b0-240.torrent">
                </media:content>
                <media:rating>nonadult</media:rating>
            </media:group>
            <media:thumbnail url="https://framatube.org/static/thumbnails/ed5c048d-01f3-4ceb-97db-6e278de512b0.jpg" height="122" width="223">
            </media:thumbnail>
```
How it is displayed:
![rssowl-peertube](https://user-images.githubusercontent.com/60163/69174301-cd922000-0b01-11ea-93f8-a021391c9960.png)

**Comment by JohnPlanetary** [comment]
I agree with these being needed! I was coming now to request the same kind of thing.

For Youtube I would like to see also some additional AI (Artificial Intelligence):

- Properly detect the dates of the published videos with the `<published>` and `<updated>` tags.
- Display the video link using either the `<yt:videoId>` tag to compose it or get it from `<link rel="alternate" href="https://www.youtube.com/watch?v=VIDEOID"/>` tag.
- Display the video on the RSSOwlnix using the embed feature and pulling the video id from the `<yt:videoId>` tag, example: https://www.youtube.com/embed/videoId?autoplay=0 So that the person can see it directly in the reader without needing to open the browser. If not possible to program that way (specially for Youtube, and maybe a few more very highly used video web sites) at least that link to the video.
- Display the video image properly with the `<media:thumbnail url="https://imageaddress" width="480" height="360"/>` tag.
- Detect the many possible different URL's used for youtube to get the RSS feed URL from the URL the user uses.
For example, if I use "https://www.youtube.com/user/beatrizmleonardoful" I will get the RSS feed tag "rssUrl" that will point to: "https://www.youtube.com/feeds/videos.xml?channel_id=UCNLr-qczxSYccaMVzit7RVg" but if I use "https://www.youtube.com/channel/UCNLr-qczxSYccaMVzit7RVg", "https://www.youtube.com/channel/UCNLr-qczxSYccaMVzit7RVg/", "https://www.youtube.com/channel/UCNLr-qczxSYccaMVzit7RVg/videos", "https://www.youtube.com/channel/UCzo1yKiVxkvFTryrsRg7KIw/channels" while the channel ID is right there in the URL the RSSOwlnix is not "smart" enough to recognize that it can get the Channel ID from the URL the user just supply to it and convert it to "https://www.youtube.com/feeds/videos.xml?channel_id=UCNLr-qczxSYccaMVzit7RVg" and then get the proper data. It would be nice for it to detect, because not all channels have the friendly name, or are almost impossible to discover.

**Comment by Xyrio** [comment]
media tags are already supported partially.
- published and updated are loaded correclty
- youbute video link is used when you double click or drag the news
- media:content links are provided as attachment downloads for youtube that links to old flash version of video which youtube  itself does not support anymore

the other stuff is ignored

**Comment by Xyrio** [comment]
added more to 2.9.0

**Comment by heartlessglasses** [comment]
Any possibility of showing info about the video length? "duration" seems to be a thing. I think being able to see the video length in (hh:)mm:ss format as a column would be ideal.

**Comment by Xyrio** [comment]
i can add it but i dont know of any site that uses it (i.e. youtube doesnt)

**Comment by heartlessglasses** [comment]
Ah, that's a bit of a shame. If I knew how to code for RSSOwl, I'd try to figure out some way to extract the duration, but I don't think I'm skilled enough to do that. Ideally though, YouTube should start using the "duration" media tag; I wonder if it'd be possible to convince them to do that, hmm...

(Also, I want to give a huge thanks to you for continuing to work on this project, it means a lot as it's still the best RSS reader out there for me.)

**Comment by Bzzz** [comment]
Hi, since this is still classified as "open", I'd like to second the enhancement request.

I've been using classic RSSOwl way back and have switched to QuiteRSS after it got abandoned. Now, about twelve years later, QuiteRSS has been abandoned as well and actually blocks the final packages on my Ubuntu 24 -> 26 LTS upgrade...and I found out that RSSOwl has been revived :)
The newspaper mode is basically all I need, and no "common" RSS reader is doing that anymore - so I'm eager to steer RSSOwlnix to where it's a full replacement for my current reader. And that requires support for the `<media:thumbnail url="https://abc" />` tag, because let's face it, this is the internet, the internet is for cat content, and ICHC (http://icanhas.cheezburger.com/rss) is using those tags for preview images. RSSOwlnix is currently not displaying them, which makes everybody sad...

Jokes aside, if I can be of help, either coding directly (I'm not very good at this...), testing proposed code changes, or posting a bounty, let me know!
